thingsboard_gateway/connectors/modbus_async/modbus_connector.py

```diff
@@ -0,0 +1,697 @@
+                                          retry_on_empty=slave.retry_on_empty,
+                                          retry_on_invalid=slave.retry_on_invalid,
+                                          retries=slave.retries,
+                                          **slave.tls)
+        elif slave.type == 'tcp':
+            master = AsyncModbusTcpClient(slave.host,
+                                          slave.port,
+                                          framer,
+                                          timeout=slave.timeout,
+                                          retry_on_empty=slave.retry_on_empty,
+                                          retry_on_invalid=slave.retry_on_invalid,
+                                          retries=slave.retries)
+        elif slave.type == 'udp':
+            master = AsyncModbusUdpClient(slave.host,
+                                          slave.port,
+                                          framer,
+                                          timeout=slave.timeout,
+                                          retry_on_empty=slave.retry_on_empty,
+                                          retry_on_invalid=slave.retry_on_invalid,
+                                          retries=slave.retries)
+        elif slave.type == 'serial':
+            master = AsyncModbusSerialClient(method=slave.method,
+                                             port=slave.port,
+                                             timeout=slave.timeout,
+                                             retry_on_empty=slave.retry_on_empty,
+                                             retry_on_invalid=slave.retry_on_invalid,
+                                             retries=slave.retries,
+                                             baudrate=slave.baudrate,
+                                             stopbits=slave.stopbits,
+                                             bytesize=slave.bytesize,
+                                             parity=slave.parity,
+                                             strict=slave.strict)
+        else:
+            raise Exception("Invalid Modbus transport type.")
+
+        return master
+
+    def __get_master_connection(self, slave: Slave):
+        """
+        Method check if connection to master already exists and return it
+        or create new connection and return it
+
+        Newly created master connection structure will look like:
+        self._master_connections = {
+            '127.0.0.1:5021': AsyncClient object
+        }
+        """
+
+        socket_str = slave.host + ':' + str(slave.port)
+        if socket_str not in self._master_connections:
+            master_connection = self.__configure_master(slave)
+            self._master_connections[socket_str] = master_connection
+
+        return self._master_connections[socket_str]
+
+    def __add_slave(self, slave_config):
+        slave_config.update({
```

Comment:
Here you can create configuration entity (That I mentioned in comments for init method in slave.py) and validate it before creating the slave itself.